CodeJam.Main.Tests/IO/TempDataTests.cs

```diff
@@ -98,6 +112,46 @@ public void TestDirectoryNestedContent()
 			Assert.IsFalse(File.Exists(nestedFile), "File should NOT exist");
 		}
 
+		[Test]
+		public void TestRetryPolicy()
+		{
+			int disposeCallCount = 0;
+			try
+			{
+				Configuration.SetTempDataRetryCallback(
+					a =>
+					{
+						Interlocked.Increment(ref disposeCallCount);
+						a();
+					});
+
+				string dirPath;
+				string nestedFile;
+				using (var dir = TempData.CreateDirectory())
+				{
+					dirPath = dir.Path;
+					nestedFile = Path.Combine(dirPath, "test.tmp");
+
+					using (File.Create(nestedFile))
+					{
+						Assert.Throws<IOException>(() => dir.Dispose());
+						Assert.AreEqual(disposeCallCount, 1);
+					}
+
+					Assert.DoesNotThrow(() => dir.Dispose());
+					Assert.AreEqual(disposeCallCount, 2);
+
+				}
+				Assert.IsFalse(Directory.Exists(dirPath), "Directory should NOT exist");
+				Assert.IsFalse(File.Exists(nestedFile), "File should NOT exist");
+				Assert.AreEqual(disposeCallCount, 2);
```

Comment:
2 should come first.
Order is expected , actual.

Reply:
Fixed)